components/link_tooltip/link_tooltip.tsx

```diff
@@ -48,21 +51,19 @@ export default class LinkTooltip extends React.PureComponent<Props, State> {
         };
     }
 
-    public showTooltip = (e: React.MouseEvent<HTMLSpanElement>): void => {
+    componentDidUpdate(prevProps: Props, prevState: State) {
         //clear the hideTimeout in the case when the cursor is moved from a tooltipContainer child to the link
         window.clearTimeout(this.hideTimeout);
 
-        if (!this.state.show) {
-            const target = e.currentTarget;
-            const tooltipContainer = this.tooltipContainerRef.current;
+        //clear the old this.showTimeout if there is any before overriding
+        window.clearTimeout(this.showTimeout);
```

Comment:
Thanks for changing that. One last thing, these `clearTimeout` calls should be inside the if block as well since `componentDidUpdate` could be triggered by some other state or props change. Generally, almost all the code inside of a `componentDidUpdate` should be wrapped in an if statement like that to avoid triggering on unexpected props/state changes

Reply:
Thanks for the feedback, `clearTimeout` calls are moved to inside of if block